server/src/main/java/org/opensearch/identity/Identity.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+/**
+ * The distinguishing character or personality of an entity within OpenSearch.
+ */
+public interface Identity {
+
+    /** A durable, reusable identifier of this identity */
+ 
+    String getId();
```

Comment:
If we're creating our own Identity object, we should have fields that we can deserialize with Jackson to follow at least the minimal user representation per [SCIM](https://www.simplecloud.info/).  From [RFC 7643](https://datatracker.ietf.org/doc/html/rfc7643#section-8.1):
```json
{
  "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
  "id": "2819c223-7f76-453a-919d-413861904646",
  "userName": "bjensen@example.com",
  "meta": {
    "resourceType": "User",
    "created": "2010-01-23T04:56:22Z",
    "lastModified": "2011-05-13T04:42:34Z",
    "version": "W\/\"3694e05e9dff590\"",
    "location":
     "https://example.com/v2/Users/2819c223-7f76-453a-919d-413861904646"
  }
}

            Figure 3: Example Minimal User JSON Representation
```
So I'd suggest at least getters for `Set<String>` schemas, `String` id, `String` userName, and `Map<String,String>` meta.

But rather than code it ourselves, we could import a dependency that implements the standard.  See http://www.simplecloud.info/#Implementations2 and scan the Open Source (AL2).  I've seen [WSO2 Charon](https://github.com/wso2/charon) used elsewhere.  [Apache Syncope](https://syncope.apache.org/) is ASF-maintained.

They might be overkill, but would provide room to grow.  I'd recommend at least taking a look at how easy it is to generate a simple identity using one of those dependencies.

Reply:
Good notion following the standard format, while I didn't include serialization in this change I am onboard that our solution should follow suite.

Your instincts to leverage an external library are on point - this is the path I'll redirect to